oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/CoreModuleProvider.java

```diff
@@ -315,6 +316,8 @@ public void start() throws ModuleStartException {
                 .registerRemote(gRPCServerInstance);
         }
 
+        OAPNodeChecker.setROLE(CoreModuleConfig.Role.valueOf(moduleConfig.getRole()));
```

Comment:
This is not a perfect way to do this. Take a look at how L304 works. We should add this as a part of that IF. 
Your way in here could cause `IllegalArgumentException` if a user sets the value as `mixed1`, rather than `Mixed`. The whole OAP will fail. If you use the better way, then it just fails back to the Mixed mode.

Reply:
Glad to receive you reply, i undersatand the problem, but i cann't understand L304 relation with it, but i find a solution and PR again, is this ok?

Reply:
I think about it， Did you mean create a new enum class to store current server role?